Docs: Document cell options by cell type (#106961)
grafana/grafana: docs/sources/panels-visualizations/visualizations/table/index.md

If you want to apply a cell type to only some fields instead of all fields, you can do so using the **Cell options > Cell type** field override.

<!-- prettier-ignore-start -->
| Cell type                                 | Description                                                                                                                |
| ----------------------------------------- | -------------------------------------------------------------------------------------------------------------------------- |

> | [Auto](#auto)                             | Automatically displays values with sensible defaults applied. |

Reviewer: maybe we should describe this as a basic text and number cell.

> | [Sparkline](#sparkline)                   | Shows values rendered as a sparkline. |

Reviewer: this is a bit weird to list second since it requires an additional transform, and data re-shaping. i would maybe put this after Gauge

Reviewer: hmm, yeah we should fix this.

| [Colored text](#colored-text)             | If thresholds are set, then the field text is displayed in the appropriate threshold color. |

> | [Colored background](#colored-background) | If thresholds are set, then the field background is displayed in the appropriate threshold color. |

Reviewer: both these types support not just Thresholds, but also Value mappings and Color schemes

> | [Gauge](#gauge)                           | Cells can be displayed as a graphical gauge, with several different presentation types. You can set the [Gauge display mode](#gauge-display-mode) and the [Value display](#value-display) options. |

Reviewer: instead of "Cells can be displayed as a graphical gauge"

maybe "Values are displayed as a horizontal bar gauge"

| Data links                                | If you've configured data links, when the cell type is **Auto**, the cell text becomes clickable. If you change the cell type to **Data links**, the cell text reflects the titles of the configured data links. To control the application of data link text more granularly, use a **Cell option > Cell type > Data links** field override. |
| [JSON View](#json-view)                   | Shows values formatted as code. |

> | [Image](#image)                           | If the field value is an image URL or a base64 encoded image, the table displays the image. |

Author: ```suggestion
| [Image](#image)                           | Displays an image when the value is a URL or a base64 encoded image. |
```

| [Actions](#actions)                       | The cell displays a button that triggers a basic, unauthenticated API call when clicked. |
<!-- prettier-ignore-end -->

#### Auto

> This cell type automatically displays values with sensible defaults applied.

Reviewer: i guess this is synced with the table above, so i wont comment on each section.

It has the following cell options:

<!-- prettier-ignore-start -->
| Option | Description |
| ------ | ----------- |

> | Wrap text | <p>Toggle the **Wrap text** switch to wrap text in the cell that contains the longest content in your table. To wrap the text _in a specific column only_, use a **Fields with name** [field override](ref:field-override), select the **Cell options > Cell type** override property, and toggle on the **Wrap text** switch.</p><p>Text wrapping is in [public preview](https://grafana.com/docs/release-life-cycle/#public-preview), however, it’s available to use by default.</p> |

Reviewer: (no edits need, just filling in some context for future):

> to wrap text in the cell that contains the longest content in your table

i think this is true for the current table but not for the one that'll be replacing it. the new one wraps text in all cells that need it based on their content and column widths, not just the longest one.

> Text wrapping is in [public preview]

this will be GA in the new table.

> | Cell value inspect | <p>Enables value inspection from table cells. When the switch is toggled on, clicking the inspect icon in a cell opens the **Inspect value** drawer which contains two tabs: **Plain text** and **Code editor**.</p><p>Grafana attempts to automatically detect the type of data in the cell and opens the drawer with the associated tab showing. However, you can switch back and forth between tabs.</p> |

Author: @leeoniya 

> > If you want to apply this setting to only some fields instead of all fields, you can do so using the **Cell options > Cell value inspect** field override.

> i feel like there are so many options that can be per-feld-overridden (in each panel) that we should just add some terse link to a dedicated section that describes how to use field overrides. and in each section here have some uniform indicator/link when the option is per-field overrideable.

I suspect this language is here because it was worth mentioning when we first added the feature. I'm gonna take it out here. As for adding something like this for every option that's configurable as an override, I'll open an issue and explore the idea. I have a sense it might be an extremely heavy lift in relation to the return for users, but if we can come up with an elegant way to do it that doesn't entail adding conceivably 50+ links to a page, then it might be worth doing.

> | Cell value inspect | <p>Enables value inspection from table cells. When the switch is toggled on, clicking the inspect icon in a cell opens the **Inspect value** drawer which contains two tabs: **Plain text** and **Code editor**.</p><p>Grafana attempts to automatically detect the type of data in the cell and opens the drawer with the associated tab showing. However, you can switch back and forth between tabs.</p><p>If you want to apply this setting to only some fields instead of all fields, you can do so using the **Cell options > Cell value inspect** field override.</p> |

Author: I suspect this language is here because it was worth mentioning when we first added the feature. I'm gonna take it out here. As for adding something like this for every option that's configurable as an override, I'll open an issue and explore the idea. I have a sense it might be an extremely heavy lift in relation to the return for users, but if we can come up with an elegant way to do it that doesn't entail adding conceivably 50+ links to a page, then it might be worth doing.

| Connect null values | How null values, which are gaps in the data, appear on the graph. Null values can be connected to form a continuous line or set to a threshold above which gaps in the data are no longer connected. Choose from:<ul><li>**Never** - Time series data points with gaps in the data are never connected.</li><li>**Always** - Time series data points with gaps in the data are always connected.</li><li>**Threshold** - Specify a threshold above which gaps in the data are no longer connected. This can be useful when the connected gaps in the data are of a known size or within a known range, and gaps outside this range should no longer be connected.</li></ul> |
| Show points         | Whether to show data points to lines or bars. Choose from: <ul><li>**Auto** - Grafana determines a point's visibility based on the density of the data. If the density is low, then points appear.</li><li>**Always** - Show the points regardless of how dense the dataset is.</li><li>**Never** - Don't show points.</li></ul> |
| Point size          | Set the size of the points, from 1 to 40 pixels in diameter. |
| Bar alignment       | Set the position of the bar relative to a data point. |

> <!-- prettier-ignore-end -->

Reviewer: is this mostly copy-paste from existing StateTimeline docs? if not i will review it. otherwise, maybe it's worth just linking out to those docs here?

<!-- prettier-ignore-start -->
| Option | Description |
| ------ | ----------- |
| Wrap text | <p>Toggle the **Wrap text** switch to wrap text in the cell that contains the longest content in your table. To wrap the text _in a specific column only_, use a **Fields with name** [field override](ref:field-override), select the **Cell options > Cell type** override property, and toggle on the **Wrap text** switch.</p><p>Text wrapping is in [public preview](https://grafana.com/docs/release-life-cycle/#public-preview), however, it’s available to use by default.</p> |
| Cell value inspect | <p>Enables value inspection from table cells. When the switch is toggled on, clicking the inspect icon in a cell opens the **Inspect value** drawer which contains two tabs: **Plain text** and **Code editor**.</p><p>Grafana attempts to automatically detect the type of data in the cell and opens the drawer with the associated tab showing. However, you can switch back and forth between tabs.</p><p>If you want to apply this setting to only some fields instead of all fields, you can do so using the **Cell options > Cell value inspect** field override.</p> |

> <!-- prettier-ignore-end -->

Author: I don't want to make folks bounce around the page so I'd say let's leave it be. We're already making sections for every cell type, so there's no point in now forcing a reader to bounce back and forth.

> <!-- prettier-ignore-end -->

Reviewer: is there any way we can create snippets and reuse them so that during review we're not having to review identical or near-identical content?

> <!-- prettier-ignore-end -->

Author: I can definitely do it for Auto and Colored background, but Colored text is tricky because there are more items in the table. I'll see what I can do.

![Table with colored background cell type](/media/docs/grafana/panels-visualizations/screenshot-table-colored-bkgrnd-v11.3-2.png)

You can also set background cell color by row:

![Table with background cell color applied to row](/media/docs/grafana/panels-visualizations/screenshot-table-colored-row-v11.3.png)

The colored background cell type has the following options:

<!-- prettier-ignore-start -->
| Option | Description |
| ------ | ----------- |
| Background display mode | Choose between **Basic** and **Gradient**. |
| Apply to entire row | Toggle the switch on to apply the background color that's configured for the cell to the whole row. |
| Wrap text | <p>Toggle the **Wrap text** switch to wrap text in the cell that contains the longest content in your table. To wrap the text _in a specific column only_, use a **Fields with name** [field override](ref:field-override), select the **Cell options > Cell type** override property, and toggle on the **Wrap text** switch.</p><p>Text wrapping is in [public preview](https://grafana.com/docs/release-life-cycle/#public-preview), however, it’s available to use by default.</p> |

> | Cell value inspect | <p>Enables value inspection from table cells. When the switch is toggled on, clicking the inspect icon in a cell opens the **Inspect value** drawer which contains two tabs: **Plain text** and **Code editor**.</p><p>Grafana attempts to automatically detect the type of data in the cell and opens the drawer with the associated tab showing. However, you can switch back and forth between tabs.</p><p>If you want to apply this setting to only some fields instead of all fields, you can do so using the **Cell options > Cell value inspect** field override.</p> |

Reviewer: any way we can just say "same as Auto cell, see above"
